### ISSUE TITLE
MHV Appts / Accessibility design pattern for button that triggers state change on page

### ISSUE THREAD

**Issue by outerpress** [department-of-veterans-affairs/va.gov-team#133972]
### 📝 Description

We'd like help developing (and possibly implementing) improvements to our patterns for changing the state of content on various pages when sorting and filtering.

### 📝 Background and context

The appointments tool has a few interactions where changing the value of a select list modifies the state of the page:
- [Past appointments list](https://www.figma.com/design/eonNJsp57eqfPqx7ydsJY9/Feature-Reference-%7C-Appointments-FE?node-id=3152-39900&t=aMoZ97dpsmaU0FRY-4)
- [Choose a location](https://www.figma.com/design/eonNJsp57eqfPqx7ydsJY9/Feature-Reference-%7C-Appointments-FE?node-id=0-2223)
- [Choose a provider](https://www.figma.com/design/eonNJsp57eqfPqx7ydsJY9/Feature-Reference-%7C-Appointments-FE?node-id=0-2428)

Collab cycle has given the team feedback that these interactions should include a trigger beyond just changing the select list to instantiate the state change. [This ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/88700) provides the most detailed feedback. ([This epic](https://github.com/department-of-veterans-affairs/va.gov-team/issues/97264) includes all related tickets)

This type of pattern is used in other tools as well, and related to sorting and filtering.

### ✋ Point of contact

Peter Russo - @outerpress

### 🛠️ Product

Appointments

### 🗣️ Slack channel

#unified-appointments-experience-ux

### ⏳ Timeframe

No specific timeframe, though we expect this feedback to come up every with staging review. 

### Evaluation Support

- [ ] Design review — Review design artifacts to identify accessibility issues and recommend solutions.
- [ ] Conformance testing — Perform an accessibility audit of a functional user flow to identify accessibility issues and recommend solutions.
- [ ] Accessibility artifact— Prepare an accessibility artifact for a Collaboration Cycle review and identify issues and recommend solutions.

### Implementation Support

- [x] Design — Prepare accessible designs, collaborate on design tasks and resolving tickets, etc.
- [x] Engineering — Prepare accessible prototypes, collaborate on engineering tasks and resolving tickets, etc.

### General Support

- [ ] Issue review — Review tickets or other reports of accessibility issues and provide guidance on how to address issues.
- [ ] Consulting — Provide general guidance and training on accessibility topics.

### 🔗 Related issues

[This epic](https://github.com/department-of-veterans-affairs/va.gov-team/issues/97264) includes all related tickets

### ✅ Tasks

- [x] Add ticket labels — Add the product's project and labels.
- [x] Save and submit — Save and submit this intake ticket.
- [x] Share on Slack — Start a new thread and post a link to this ticket in the mhv-accessibility Slack channel
